src/action/base/create_or_insert_file.rs

```diff
@@ -0,0 +1,229 @@
+use nix::unistd::{chown, Group, User};
+
+use crate::action::{Action, ActionDescription, ActionError, StatefulAction};
+use rand::Rng;
+use std::{
+    io::SeekFrom,
+    os::unix::prelude::PermissionsExt,
+    path::{Path, PathBuf},
+};
+use tokio::{
+    fs::{remove_file, OpenOptions},
+    io::{AsyncReadExt, AsyncSeekExt, AsyncWriteExt},
+};
+use tracing::{span, Span};
+
+/** Create a file at the given location with the provided `buf`,
+optionally with an owning user, group, and mode.
+
+If the file exists, the provided `buf` will be inserted at its beginning.
+ */
+#[derive(Debug, serde::Deserialize, serde::Serialize, Clone)]
+pub struct CreateOrInsertFile {
```

Comment:
I wonder if, since this creates the content at the start of the file, we should call it `CreateOrPrependFile`?

Reply:
I have no preference. If you prefer that I'll change it :)

Reply:
Could I ask you to? :) Otherwise could we give this an enum arg called `Position` which has `Prepend`, `Append`?